### PR TITLE
dockerize snapshot upload

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
**Summary of changes**
Changes introduced in this pull request:
- dockerized snapshot service, which can now be run completely locally with only Docker as a requirement

I'll pin the snapshot version in the TF to a specific version once it is built by CI (it needs to reach `main` first).

**Reference issue to close (if applicable)**

Part of https://github.com/ChainSafe/forest-iac/issues/364

**Other information and links**

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
@hanabi1224 @jdjaustin

**Review by samuelarogbonlo — APPROVED** [review]
LGTM

File: .github/workflows/snapshot-service-image.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        with:
+          images: ghcr.io/chainsafe/forest-snapshot-service
+          tags: |
+            type=raw,value={{date 'YYYY-MM-DD'}}-{{sha}}
+            type=raw,value=latest,enable={{is_default_branch}}
+
+      - name: Build image and push to GitHub Container Registry
+        uses: docker/build-push-action@v5
+        with:
+          context: ./images/snapshot-service/
```
Comment (hanabi1224) [inline .github/workflows/snapshot-service-image.yml:49]:
```suggestion
          context: ./images/snapshot-service/
          load: true
```

Comment (hanabi1224) [reply .github/workflows/snapshot-service-image.yml:49]: I believe without `load: true`, `docker image ls` steps won't list the output image

File: images/snapshot-service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    gem install \
+        docker-api:1.28.0 \
+        slack-ruby-client:2.2 \
+        activesupport:7.0.8
+
+COPY ./src /opt/snapshot-service
+
+# TODO: Change this once `sync-check` is fully-dockerized as well.
+# hadolint ignore=DL3022
+COPY --from=common ruby_common /opt/snapshot-service/ruby_common
```
Comment (hanabi1224) [inline images/snapshot-service/Dockerfile:21]: where is `common` defined?

Comment (hanabi1224) [reply images/snapshot-service/Dockerfile:21]: I see, it's from `build-context`. Maybe adding some comments to reduce the confusion

**Review by hanabi1224 — APPROVED** [review]

File: .github/workflows/snapshot-service-image.yml (hunk 1 above)
Comment (hanabi1224) [reply .github/workflows/snapshot-service-image.yml:49]: Nvm it actually does. https://github.com/ChainSafe/forest-iac/actions/runs/7726607692/job/21063223556?pr=400#step:7:30

File: images/snapshot-service/Dockerfile (hunk 2 above)
Comment (LesnyRumcajs) [reply images/snapshot-service/Dockerfile:23]: @hanabi1224 done. I had to drop `apk` packages pinning, sadly, alpine repos don't keep old packages, so there's no way to meaningfully pin the versions.

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**